src/nbl/ui/CWindowXCB.cpp

```diff
@@ -0,0 +1,296 @@
+    };
+
+    xcb_void_cookie_t xcbCheckResult = xcb.pxcb_create_window(
+        m_connection->getRawConnection(), XCB_COPY_FROM_PARENT, m_xcbWindow, primaryScreen->root,
+        static_cast<int16_t>(m_x),
+        static_cast<int16_t>(m_y),
+        static_cast<int16_t>(m_width),
+        static_cast<int16_t>(m_height), 4,
+        XCB_WINDOW_CLASS_INPUT_OUTPUT, primaryScreen->root_visual, eventMask,
+        valueList);
+
+    setWindowSize_impl(m_width, m_height);
+    
+    auto WM_DELETE_WINDOW = m_connection->resolveAtom(m_WM_DELETE_WINDOW);
+    auto NET_WM_PING = m_connection->resolveAtom(m_NET_WM_PING);
+    auto WM_PROTOCOLS = m_connection->resolveAtom(m_WM_PROTOCOLS);
+    
+    const std::array atoms {WM_DELETE_WINDOW, NET_WM_PING};
+    xcb.pxcb_change_property(
+            m_connection->getRawConnection(), 
+            XCB_PROP_MODE_REPLACE, 
+            m_xcbWindow, 
+            WM_PROTOCOLS, XCB_ATOM_ATOM, 32, atoms.size(), atoms.data());
+
+
+    auto motifHints = fetchMotifMWHints(getFlags().value);
+    m_connection->setMotifWmHints(m_xcbWindow, motifHints);
+    
+    if(isAlwaysOnTop()) {
+        XCBConnection::XCBAtomToken<core::StringLiteral("NET_WM_STATE_ABOVE")> NET_WM_STATE_ABOVE;
+        m_connection->setNetMWState(
+            primaryScreen->root,
+            m_xcbWindow, false, m_connection->resolveAtom(NET_WM_STATE_ABOVE));
+    }
+
+    xcb.pxcb_map_window(m_connection->getRawConnection(), m_xcbWindow);
+    xcb.pxcb_flush(m_connection->getRawConnection());
+    m_dispatcher.start();
+}
+
+CWindowXCB::~CWindowXCB()
+{
+}
+
+IClipboardManager* CWindowXCB::getClipboardManager() {
+    return m_clipboardManager.get();
+}
+
+ICursorControl* CWindowXCB::getCursorControl() {
+    return m_cursorControl.get();
+}
+
+IWindowManager* CWindowXCB::getManager() {
+    return m_windowManager.get();
+}
+
+bool CWindowXCB::setWindowSize_impl(uint32_t width, uint32_t height) {
+    auto& xcb = m_windowManager->getXcbFunctionTable();
+    auto& xcbIccm = m_windowManager->getXcbIcccmFunctionTable();
+
+    xcb_size_hints_t hints = {0};
+
+    xcbIccm.pxcb_icccm_size_hints_set_size(&hints, true, width, height);
+    if(!isResizable()) {
+        xcbIccm.pxcb_icccm_size_hints_set_min_size(&hints, width, height);
+        xcbIccm.pxcb_icccm_size_hints_set_max_size(&hints, width, height);
+    }
+    xcbIccm.pxcb_icccm_set_wm_normal_hints(m_connection->getRawConnection(), m_xcbWindow, &hints);
+    return  true;
+}
+
+bool CWindowXCB::setWindowPosition_impl(int32_t x, int32_t y) {
+    auto& xcb = m_windowManager->getXcbFunctionTable();
+
+    const int32_t values[] = { x, y };
+    auto cookie = xcb.pxcb_configure_window_checked(m_connection->getRawConnection(), m_xcbWindow, XCB_CONFIG_WINDOW_X | XCB_CONFIG_WINDOW_Y, values);
+    bool check = checkXcbCookie(xcb, m_connection->getRawConnection(), cookie);
+    xcb.pxcb_flush(m_connection->getRawConnection());
+    assert(check);
+    return true;
```

Comment:
do not asert, this is why the function returns a bool, so it can handle failures